website/blog/2024-04-22-extended-attributes.md

```diff
@@ -0,0 +1,103 @@
+
+![Extended Attributes text box](/img/blog/2024-04-10-extended-attributes/ext_attr.png)
+
+_<small>At the bottom of an Environment's Settings you’ll find the Extended Attributes text box.</small>_
+
+### Development
+
+```yaml
+account: 123dev
+project: dev
+dataset: dbt_winnie
+method: oauth
+threads: 1
+```
+
+### Staging
+
+```yaml
+account: 123dev
+project: staging
+dataset: main
+method: service-account-json
+threads: 16
+```
+
+### Production
+
+```yaml
+account: 456prod
+project: analytics
+dataset: main
+method: service-account-json
+threads: 16
+```
+
+With this setup, we have a separate account for development work, using individual development datasets for each developer (with a single thread so that the development build logs are easier to read) connected via OAuth, and a shared `staging` project with a default `main` dataset for the Staging environment that is only built to via a GCP Service Account through dbt Cloud. In that project we can then configure IAM permissions to only allow building into the Staging schema from jobs that use the Staging environment as well.
+
+Production is then pointed to a _completely separate account_ that is only writable from Production environment builds and readable from the BI tool.
+
+It’s really that simple. This works with PrivateLink connections handling the authentication as well! Again, while we have 1 project connection that is the _default_, you can now configure unique connections securely _per Environment_.
+
+## All the world a Stage
+
+We touched on Staging Environments above in discussing Extended Attributes, but let's dig deeper into how dbt Cloud now supports those in a first class way. You now have the option when configuring an Environment to choose Development, Production, _or_ Staging. When you can configure an Environment as a Staging type, you’ll unlock new abilities, most importantly the ability to defer to _that_ Environment for Development work. This fully enables a proper Write-Audit-Publish flow, where Development work is built against and promoted to Staging before being merged into a Production branch when releases have been tested.
+
+All you need to do is configure an Environment as Staging, and toggle the control in the dbt Cloud IDE to “Defer to staging/production”. This will favor a Staging environment over Prod if you have one set up.
```

Comment:
```suggestion
All you need to do is configure an environment as staging and enable the **Defer to staging/production** option in the dbt Cloud IDE. Doing this will favor a staging environment over prod if you have one set up.
```